docker/templates/macros.m4

```diff
@@ -3,8 +3,10 @@ m4_divert(-1)
 m4_define(
     `apt_install',
     `m4_dnl
-apt-get update && \
-    apt-get install --no-install-suggests --no-install-recommends -y $1 && \
+apt-get update && ( \
+    apt-get install --no-install-suggests --no-install-recommends -y $1 || ( \
+        apt-key adv --keyserver hkps.pool.sks-keyservers.net --refresh-keys && \
```

Comment:
Any reason we don't always run this first, instead of only inside an `|| ()`? (If the first `apt-get install` fails, it's not necessarily the keys that were an issue either.)

Reply:
Cause it is unneeded in most cases. Depending on the key server drawn from the pool, refreshing keys might take some time. Retrying after refreshing keys won’t hurt even though keys might not be the issue. 

The only case of failure we have is with yarn package. Would you prefer systematically refreshing yarn keys only? See https://github.com/metabrainz/musicbrainz-docker/commit/2906242e6c5063022b0c6737d8462c109de57b47 for example, which could apply to https://github.com/metabrainz/musicbrainz-server/blob/c6ddafd6cf4e9b6f7d5ea0e1bafc6de809ad8e3f/docker/templates/macros.m4#L25

Reply:
If refreshing only the yarn keys is fast, I don't see an issue with doing it unconditionally, otherwise what you have works too.

Reply:
It is not really faster than refreshing all keys, as the delay is usually due to server latency/load. I stick with what I have.